fms_dgt/base/task.py

```diff
@@ -91,9 +92,12 @@ def __init__(
         if dataloader is None:
             self._dataloader = DefaultDataloader(data=seed_examples)
         else:
-            assert (
-                TYPE_KEY in dataloader
-            ), f"Must specify data loader type with '{TYPE_KEY}' key"
+            assert TYPE_KEY in dataloader, (
+                "Must specify dataloader type with %s key",
+                TYPE_KEY,
+            )
+            if dataloader[TYPE_KEY] == "file":
+                dataloader["seed_examples"] = seed_examples
```

Comment:
I think just having seed_examples assumed to be a kwarg of dataloader would suffice here, we don't want to add logic for a particular dataloader in this task class

```suggestion
assert TYPE_KEY in dataloader, (
                "Must specify dataloader type with %s key",
                TYPE_KEY,
            )
self._dataloader = get_dataloader(dataloader.pop(TYPE_KEY))(**dataloader)
```

Reply:
I'm a little confused. I get that we don't want a dataloader-specific logic inside the task class, but I don't see `seed_examples` as part of dataloader kwarg in your suggestion..

Reply:
More what I'm thinking is like how we do it for datastores [here](https://github.com/foundation-model-stack/fms-dgt/blob/970a211939f25d3b9f3c533b84a5860c8be30bb2/fms_dgt/base/task.py#L74). Just have a dict 

`dl_kwargs = { 'seed_examples' : seed_examples, ... }`

And that's passed in like the datastore args like

`**{**dl_kwargs, **dataloader}`

Then it's applied no matter the dataloader

Reply:
In fact, should we also separate out the initialization of datastores and dataloaders (each individually), just putting those functions into internal methods, so people can overwrite as needed?

Reply:
Ok I updated it! And yeah, we could do that. But it's going to take a bit more refactoring as we need to separate out the kwargs needed for initializing datastores and dataloaders so maybe do that in a separate PR?